public/app/plugins/datasource/loki/querybuilder/parsing.ts

```diff
@@ -182,22 +182,24 @@ function getLabel(expr: string, node: SyntaxNode): QueryBuilderLabelFilter {
 }
 
 function getLineFilter(expr: string, node: SyntaxNode): { operation?: QueryBuilderOperation; error?: string } {
-  // Check for nodes not supported in visual builder and return error
-  const ipLineFilter = getAllByType(expr, node, 'Ip');
-  if (ipLineFilter.length > 0) {
+  const filter = getString(expr, node.getChild('Filter'));
+  const filterExpr = handleQuotes(getString(expr, node.getChild('String')));
+  const ipLineFilter = node.getChild('FilterOp')?.getChild('Ip');
+
+  if (ipLineFilter) {
     return {
-      error: 'Matching ip addresses not supported in query builder',
+      operation: {
+        id: LokiOperationId.LineFilterIpMatches,
+        params: [filter, filterExpr],
+      },
     };
   }
-
   const mapFilter: any = {
-    '|=': '__line_contains',
-    '!=': '__line_contains_not',
-    '|~': '__line_matches_regex',
-    '!~': '"__line_matches_regex"_not',
+    '|=': LokiOperationId.LineContains,
```

Comment:
I noticed that in `parsing.ts` I have hardcoded some values instead of using `LokiOperationId`. I have added  replacing hardcoded values to LokiOperationId enum as a possible task for intern